src/core/s3_processor.py

```
@@ -0,0 +1,58 @@
+    
+    def list_input_audio_files(self, path: str = "", extensions: Optional[List[str]] = None) -> List[str]:
+
+        if extensions is None:
+            extensions = self.config.supported_extensions
+
+        return self.list_dir(self.input_bucket, path, extensions)
+    
+    def list_results_jsons(self, path: str) -> list[str]:
+        return self.list_dir(self.output_bucket, path, ['.json'])
+    
+    def download_audio(self, file_path) -> tuple[np.ndarray, int]:
```

Ruff (0.12.2)

Missing type annotation for function argument `file_path`

(ANN001)